app/src/main/java/com/duckduckgo/app/browser/BrowserViewModel.kt

```diff
@@ -129,11 +129,19 @@ class BrowserViewModel(
     }
 
     suspend fun onNewTabRequested(isDefaultTab: Boolean = false): String {
-        return tabRepository.add(isDefaultTab = isDefaultTab)
+        return tabRepository.addWithSource(
+                isDefaultTab = isDefaultTab,
+                sourceTabId = tabRepository.liveSelectedTab.value?.tabId
```

Comment:
Related to my comment about moving `tabRepository.liveSelectedTab.value?.tabId` inside repository.